atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBoundStore.java

```diff
@@ -105,11 +105,25 @@ public Long apply(Client client) {
                     cas(client, null, CassandraTimestampUtils.INITIAL_VALUE);
                     return CassandraTimestampUtils.INITIAL_VALUE;
                 }
-                Column column = result.getColumn();
-                currentLimit = PtBytes.toLong(column.getValue());
+                currentLimit = extractUpperLimit(result);
                 DebugLogger.logger.info("[GET] Setting cached timestamp limit to {}.", currentLimit);
                 return currentLimit;
             }
+
+            private long extractUpperLimit(ColumnOrSuperColumn result) {
+                try {
+                    Column column = result.getColumn();
+                    return PtBytes.toLong(column.getValue());
+                } catch (IllegalArgumentException e) {
+                    String msg = "Caught an IllegalArgumentException trying to convert the stored value to a long. "
+                            + "This can happen if you attempt to run AtlasDB with a leader block after having "
```

Comment:
`without a timelock block` instead. We can run without either of them and hit the error.